### DISCUSSION TITLE
a created invoice change if backoffice info changes

### DISCUSSION THREAD

**Discussion by kintaro1981** [baimard/gestion#417] [Features request]
**Describe the bug**
invoice change if I change the product

**To Reproduce**
1. Create product01 with a price of 20€
2. Create an invoiceXX with 1x produtc01
3. Go to product01 and change the price to 30€
4. Go to invoiceXX and you can see that now the product01 costs 30€ so the total of the invoice changed.

**Expected behavior**
the invoice once done doesn't have to be modified in case products info, customer/clients infos, VAT amount, etc changes.

**Comment by baimard** [comment]
PLanned in V3

**Comment by baimard** [comment]
Hello,

Thank you for reporting this issue.

This behavior has already been reviewed.

The current design of Gestion keeps a link between the invoice and the associated product. As a result, modifying an existing product may also affect documents that already reference it.

For now, if the price or other characteristics of a product change, we recommend **creating a new product instead of modifying the existing one**, especially when the existing product has already been used in an invoice.

However, we agree with the underlying issue: **an invoice that has already been issued should not change afterward** because a product used in it was modified.

We are therefore currently considering a change that would prevent a product from being modified once it has been included in an invoice. This would preserve the integrity of existing invoices while keeping the current product-based architecture of Gestion.

The same principle is also being considered more broadly for information that must remain consistent once an invoice has been issued.

Thank you for your report and your suggestions. They are very helpful in improving Gestion.
